src/index.js

```diff
@@ -1,36 +1,147 @@
-    li.style.display = 'block';
-    li.style.padding = '10px 0px 10px 50px';
-    li.style.marginLeft = '-40px';
-    li.style.marginRight = '50vw';
+import './style.css';
+
+const taskInput = document.getElementById('input');
+const addButton = document.getElementById('button');
+const todoList = document.getElementById('todo-list');
+const completeButton = document.querySelector('.complete');
+
+const savedTasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+/* eslint-disable */
 
+function renderTasks() {
+  savedTasks.forEach((task) => {
+    const li = document.createElement('li');
     const checkbox = document.createElement('input');
     checkbox.type = 'checkbox';
     checkbox.checked = task.completed;
     li.appendChild(checkbox);
-    li.innerHTML += task.description;
+    li.innerHTML += `${task.text} <button class="remove-button">X</button> <button class="edit-button">Edit</button>`;
     todoList.appendChild(li);
+
+    const removeButton = li.querySelector('.remove-button');
+    removeButton.addEventListener('click', () => {
+      removeTask(li);
+    });
+
+    const editButton = li.querySelector('.edit-button');
+    editButton.addEventListener('click', () => {
+      editTask(li);
+    });
+  });
+}
+renderTasks(); // edite
+
+// save tassk
+
+function saveTasks() {
+  localStorage.setItem('tasks', JSON.stringify(savedTasks));
+}
+
+function addTask() {
+  const taskValue = taskInput.value.trim();
+
+  if (taskValue === '') {
+    return;
+  }
+
+  const task = {
+    text: taskValue,
+    completed: false,
+    index: savedTasks.length // set the index property to the new array length
+  };
+
+  savedTasks.push(task);
+  saveTasks();
+
+  const li = document.createElement('li');
+  const checkbox = document.createElement('input');
+  li.style.listStyle = 'none';
+  checkbox.type = 'checkbox';
+  li.appendChild(checkbox);
+
+  li.innerHTML += `${task.text} <button class="remove-button">X</button> <button class="edit-button">Edit</button>`;
+  todoList.appendChild(li);
+
+  const removeButton = li.querySelector('.remove-button');
+  removeButton.addEventListener('click', () => {
+    removeTask(li);
   });
-});
+
+  const editButton = li.querySelector('.edit-button');
+  editButton.addEventListener('click', () => {
+    editTask(li);
+  });
+
+  taskInput.value = '';
+}
+
+// complete button
+
+completeButton.addEventListener('click', removeCompletedTasks);
+
+// remove task
+
+function removeTask(li) {
+  const index = Array.prototype.indexOf.call(todoList.children, li);
+  savedTasks.splice(index, 1);
+  saveTasks();
+
+  // Update indexes of remaining tasks
+  for (let i = index; i < todoList.children.length; i++) {
+    const taskLi = todoList.children[i];
+    const checkbox = taskLi.querySelector('input[type="checkbox"]');
+    checkbox.setAttribute('data-index', i);
+  }
+
+  if (li.firstChild.nextSibling.checked) {
+    li.style.display = 'none';
+  } else {
+    todoList.removeChild(li);
+  }
+}
```

Comment:
- [ ] @SAZZAD1Q2 great work implementing the changes requested, however the index still seems not to update when a user deletes a todo item, below screenshot was done after adding four items deleting the first  2 and then adding a new one, you can see the indexes of the items
![image](https://user-images.githubusercontent.com/61408860/232199479-e6c3721f-1c41-41be-9930-04b6f9dec6fb.png)